File: content/problems/2325-decode-the-message/solutions/first-occurrence-substitution-table/Solution.java
class Solution {
    public String decodeMessage(String key,String message){
        char[] mapping=new char[26];
        char next='a';
        for(char ch:key.toCharArray())if(ch!=' '&&mapping[ch-'a']==0)mapping[ch-'a']=next++;
        StringBuilder answer=new StringBuilder();
        for(char ch:message.toCharArray())answer.append(ch==' '?ch:mapping[ch-'a']);
        return answer.toString();
    }
}

File: content/problems/2325-decode-the-message/solutions/first-occurrence-substitution-table/Solution.kt
class Solution {
    fun decodeMessage(key:String,message:String):String{
        val mapping=CharArray(26);
        var next='a';
        for(ch in key)if(ch!=' '&&mapping[ch-'a'].code==0)mapping[ch-'a']=next++;
        return buildString{
            for(ch in message)append(if(ch==' ')ch else mapping[ch-'a'])
        }
    }
}
